jetty-server/src/main/java/org/eclipse/jetty/server/SensingConnectionFactory.java

```diff
@@ -0,0 +1,55 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.server;
+
+import java.nio.ByteBuffer;
+import java.util.Iterator;
+
+import org.eclipse.jetty.io.Connection;
+import org.eclipse.jetty.io.EndPoint;
+
+public interface SensingConnectionFactory extends ConnectionFactory
+{
+    enum SensingOutcome
+    {
+        RECOGNIZED,
+        NOT_RECOGNIZED,
+        NEED_MORE_BYTES
+    }
+
+    SensingOutcome sense(ByteBuffer buffer);
+
+    Connection newConnection(Connector connector, EndPoint endPoint, ConnectionFactory next);
```

Comment:
Remove this method, just keep the one from `ConnectionFactory`.

Reply:
I agree that a second `newConnection` is rather confusing, but there is still the need for a mechanism to pass the `ConnectionFactory` to upgrade to when detection succeeds.

I'm going to change this in my next push, but we'll probably need to discuss this more in depth.